firmware/src/main.cpp

```diff
@@ -1,99 +1,58 @@
+    .pin_therm_motor2       = A3,
+    .pin_therm_optical      = A4,
+    // Gantry X-Axis Pins
+    .pin_motor_x_step       = 5,
+    .pin_motor_x_dir        = 6,
+    .pin_motor_x_enc_a      = 7,
+    .pin_motor_x_enc_b      = 8,
+    .pin_motor_x_ls_home    = 9,
+    .pin_motor_x_ls_far     = 10,
+    // Gantry Y-Axis Pins
+    .pin_motor_y_step       = 22,
+    .pin_motor_y_dir        = 23,
+    .pin_motor_y_enc_a      = 24,
+    .pin_motor_y_enc_b      = 25,
+    .pin_motor_y_ls_home    = 26,
+    .pin_motor_y_ls_far     = 27
+};
+
+mPMTTestStand test_stand(io);
+
+uint32_t blink_start;
+bool blink_state;
 
 void setup()
 {
-    // put your setup code here, to run once:
-    pinMode(LED_BUILTIN, OUTPUT);
-    // thermistor pin configurations
-    pinMode(thermistor1Pin, INPUT);
-    pinMode(thermistor2Pin, INPUT);
-    pinMode(thermistor3Pin, INPUT);
-    pinMode(thermistor4Pin, INPUT);
-    pinMode(thermistor5Pin, INPUT);
-
     DEBUG_INIT;
-    serial_device.ser_connect(COMM_BAUD_RATE);
-
-    setup_axis(&axis_x_config, &axis_x);
-    setup_axis(&axis_y_config, &axis_y);
-
-    analogReadResolution(12); //enable 12 bit resolution mode in Arduino Due. Default is 10 bit.
-    Thermistor10k thermistor_ambient(thermistor1Pin);
-    Thermistor10k thermistor_motor1(thermistor2Pin);
-    Thermistor10k thermistor_mpmt(thermistor3Pin);
-    Thermistor10k thermistor_motor2(thermistor4Pin);
-    Thermistor10k thermistor_optical_box(thermistor5Pin);  
-    //example usage: double temp1 = thermistor_ambient.readTemperature(); 
-}
-
-void handle_home()
-{
-    
-}
-
-void handle_move()
-{
-    uint32_t accel, hold_vel, dist;
-    uint8_t axis, dir;
-
-    uint8_t *data = comm.received_message().data;
-
-    accel    = NTOHL(data);
-    hold_vel = NTOHL(data + 4);
-    dist     = NTOHL(data + 8);
-
-    axis = data[12];
-    dir = data[13];
+    pinMode(LED_BUILTIN, OUTPUT);
 
-    Axis *axis_ptr = (axis == AXIS_X ? &axis_x : &axis_y);
-    VelProfile profile;
-    generate_vel_profile(accel, axis_ptr->vel_min, hold_vel, dist, &profile);
+    test_stand.setup();
 
-    axis_trapezoidal_move_rel(axis_ptr, profile.counts_accel, profile.counts_hold, profile.counts_decel, (Direction)dir);
-    
-    // TODO delete this log message:
-    comm.log(LL_INFO, "cts_a = %d, cts_h = %d, dist = %d, axis = %c, dir = %s",
-        profile.counts_accel,
-        profile.counts_hold,
-        dist,
-        (axis == AXIS_X ? 'x' : 'y'),
-        (dir == DIR_POSITIVE ? "pos" : "neg"));
+    blink_start = millis();
+    blink_state = false;
 }
 
-int count = 0;
-
 void loop()
 {
-    if (comm.check_for_message()) {
-        switch (comm.received_message().id) {
-            case MSG_ID_HOME:
-                break;
-            case MSG_ID_MOVE:
-                handle_move();
-                break;
-        }
+    test_stand.execute();
+
+    // Blink an LED
+    if ((millis() - blink_start) >= 500) {
+        blink_state = !blink_state;
```

Comment:
What is the meaning of `blink_state = !blink_state`?

Reply:
Toggle the `blink_state` (i.e. set it to the opposite of what it was before)